dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+        final List<Map<String, Object>> results =
+                dc.setSQL(selectActionsAndSteps()).loadObjectResults();
+
+        results.stream().forEach(row -> {
+
+            dc.setSQL(insertActionsAndSteps());
+            dc.addParam(row.get("action_id"));
+            dc.addParam(row.get("step_id"));
+            dc.addParam(row.get("my_order"));
+
+            Logger.debug(this, "Adding to workflow_action_step the row: " + row);
+
+            try {
+                dc.loadResult();
+            } catch (DotDataException e) {
+
+                Logger.error(this, "ERROR on Adding to workflow_action_step the row: "
+                        + row + ", err:" + e.getMessage(), e);
+                throw new DotRuntimeException(
+                        "An error occurred when adding data to the 'workflow_action_step' table.",
+                        e);
+            }
+        });
+    } // addWorkflowActionStepData.
+
+    private void addRequiresCheckoutOptionColumn(final DotConnect dc) throws DotDataException {
+
+        boolean needToCreate = false;
+        Logger.info(this, "Adding new 'requires_checkout_option' column to 'workflow_action' table.");
+
+        try {
+            dc.setSQL(findRequiresCheckoutOptionsColumn()).loadObjectResults();
+        } catch (Throwable e) {
+
+            Logger.info(this, "Column 'workflow_action.requires_checkout_option' does not exists, creating it");
+            needToCreate = true;
+        }
+
+        if (needToCreate) {
+            try {
+                dc.executeStatement(addRequiresCheckoutOptionColumn());
+            } catch (SQLException e) {
+                throw new DotRuntimeException("The 'requires_checkout_option' column could not be created.", e);
+            }
+        }
+    } // addRequiresCheckoutOptionColumn.
+
+    private void addSchemeIdColumn(final DotConnect dc) throws DotDataException {
+
+        boolean needToCreate = false;
+        Logger.info(this, "Adding new 'scheme_id' column to 'workflow_action' table.");
+
+        try {
+            dc.setSQL(findSchemeIdColumn()).loadObjectResults();
+        } catch (Throwable e) {
+
+            Logger.info(this, "Column 'workflow_action.scheme_id' does not exists, creating it");
+            needToCreate = true;
+        }
+        if (needToCreate) {
+            try {
+                dc.executeStatement(addSchemeIdColumn());
+            } catch (SQLException e) {
+                throw new DotRuntimeException("The 'scheme_id' column could not be created.", e);
+            }
+        }
+    } // addSchemeIdColumn.
+
+    private void createWorkflowActionStepTable(final DotConnect dc) throws DotDataException {
+
+        boolean needToCreate = false;
+        Logger.info(this, "Creating intermediate 'workflow_action_step' table.");
+
+        try {
+
+            dc.setSQL(findIntermediateTable()).loadObjectResults();
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)